lib/src/revset.rs

```diff
@@ -380,6 +380,15 @@ impl RevsetExpression {
         })
     }
 
+    /// Ancestors of `self`, including `self` up to `generations` back.
+    pub fn ancestors_at(self: &Rc<RevsetExpression>, generations: u64) -> Rc<RevsetExpression> {
+        Rc::new(RevsetExpression::Ancestors {
+            heads: self.clone(),
+            generation: (generations - 1)..generations,
```

Comment:
Perhaps, this should be `generation..(generation + 1)`. `Ancestors { generation: 0 }` is the `heads` themselves. And maybe we can remove `assert!(amount >= 1)` from the next commit.

Reply:
done